src/HederaOpenDEX/pages/Pool/reducers/poolReducer.tsx

```diff
@@ -0,0 +1,83 @@
+import { ActionType, PoolActions } from "./actionTypes";
+
+export interface PoolState {
+  inputToken: {
+    symbol: string;
+    amount: number;
+    displayedAmount: string;
+    address: string;
+    spotPrice: number;
+    balance: number | undefined;
+  };
+  outputToken: {
+    symbol: string;
+    amount: number;
+    displayedAmount: string;
+    address: string;
+    spotPrice: number;
+    balance: number | undefined;
+  };
+}
+
+const initialPoolState: PoolState = {
+  inputToken: {
+    symbol: "",
+    amount: 0.0,
+    displayedAmount: "0.0",
+    address: "",
+    spotPrice: 0,
+    balance: undefined,
+  },
+  outputToken: {
+    symbol: "",
+    amount: 0.0,
+    displayedAmount: "0.0",
+    address: "",
+    spotPrice: 0,
+    balance: undefined,
+  },
+};
+
+function initPoolReducer(initialPoolState: PoolState) {
```

Comment:
Consider using arrow function e.g 
`const initPoolReducer = (initialPoolState: PoolState): PoolState =>  initialPoolState `

It's one liner.